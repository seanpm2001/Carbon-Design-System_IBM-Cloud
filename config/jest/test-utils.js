import React from 'react'
import {render} from '@testing-library/react'
import { Theme } from '@carbon/react';
import CarbonForCloudWrapper from "../../src/i18n/CarbonForCloudWrapper";


// eslint-disable-next-line react/prop-types
const TestWrapper = ({ children }) => {
  return (
    <Theme theme="white">
      <CarbonForCloudWrapper locale='en'>
        {children}
      </CarbonForCloudWrapper>
    </Theme>
  )
}

const customRender = (ui, options) =>
  render(ui, {wrapper: TestWrapper, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {customRender as render}