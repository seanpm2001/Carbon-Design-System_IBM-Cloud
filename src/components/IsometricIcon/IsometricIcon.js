import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

// Icons
import EMPTY from "./icons/Empty";
import ERROR from "./icons/Error";
import NORESULTS from "./icons/NoResults";
import NOTIFICATION from "./icons/Notification";
import TAG from "./icons/Tag";
import UNAUTHORIZED from "./icons/Unauthorized";

const iconObj = {
  EMPTY,
  ERROR,
  NORESULTS,
  NOTIFICATION,
  TAG,
  UNAUTHORIZED,
};

const IsometricIcon = ({ id, icon, className }) => {
  const iconProps = {
    id,
    className: classnames("pal--isometric-icon", className),
  };

  const Icon = iconObj[icon];

  return <Icon {...iconProps} />;
};

IsometricIcon.defaultProps = {
  icon: "EMPTY",
  className: null,
};

IsometricIcon.propTypes = {
  id: PropTypes.string.isRequired,
  icon: PropTypes.oneOfType([
    PropTypes.oneOf([
      "EMPTY",
      "ERROR",
      "NORESULTS",
      "NOTIFICATION",
      "TAG",
      "UNAUTHORIZED",
    ]),
    PropTypes.node,
  ]),
  className: PropTypes.string,
};

export default IsometricIcon;
