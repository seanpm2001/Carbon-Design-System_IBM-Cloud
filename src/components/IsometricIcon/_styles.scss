@use "@carbon/react/scss/type";
@use '@carbon/react/scss/reset';

// @use '@carbon/layout';
// @use "@carbon/react/scss/components/button";
@use "@carbon/react/scss/components/link";
@use "@carbon/react/scss/components/overflow-menu";

// @use "@carbon/react/scss/grid";
@use "@carbon/react/scss/components/skeleton-styles";
@import "@carbon/react/scss/colors";
@import "@carbon/react/scss/spacing";
@import "@carbon/react/scss/themes";
@import "@carbon/react/scss/theme";
@import "@carbon/react/scss/config";
@import "@carbon/react/scss/components/button";
@import "@carbon/react/scss/grid";

.pal--isometric-icon {
  height: 128px;
  width: 128px;
}
