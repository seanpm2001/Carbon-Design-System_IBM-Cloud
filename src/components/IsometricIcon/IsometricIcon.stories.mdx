import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
import IsometricIcon from '.';

import './_styles.scss';

<Meta
  title="Components/IsometricIcon"
  component={IsometricIcon}
  argTypes={{
    locale: { control: { type: 'select' }, options: ['de', 'en', 'es', 'fr', 'it', 'ja', 'ko', 'pt-br', 'zh-cn', 'zh-tw'], defaultValue: 'en' },
    children: { control: false },
  }}
  parameters={{ status: { type: 'core' }}}
/>

# IsometricIcon

<Canvas>
  <Story
    name="Demo"
  >
    <div>
      <IsometricIcon id="icon-empty-example" icon="EMPTY" />
      <IsometricIcon id="icon-error-example" icon="ERROR" />
      <IsometricIcon id="icon-noresults-example" icon="NORESULTS" />
      <IsometricIcon id="icon-notification-example" icon="NOTIFICATION" />
      <IsometricIcon id="icon-tag-example" icon="TAG" />
      <IsometricIcon id="icon-unauthorized-example" icon="UNAUTHORIZED" />
    </div>
  </Story>
</Canvas>

## Description

IsometricIcon is a component used for... (FILL ME IN)

## Usage

1.  Install the [@console/pal](https://github.ibm.com/ibmcloud/pal#how-to-install-and-use) library
2.  Import the styles:

    ```js
    @import 'carbon-for-ibm-cloud/Components/IsometricIcon/styles';
    ```

3.  Import the component:

    ```js
    import { IsometricIcon } from 'carbon-for-ibm-cloud/Components';
    ```

4.  Render the component.

For general component usage review the [Components Docs](https://pages.github.ibm.com/ibmcloud/pal/development-guidelines/components/overview).

## Props

<ArgsTable of={IsometricIcon} />

## Contributors

- Ryan O. Mackey
