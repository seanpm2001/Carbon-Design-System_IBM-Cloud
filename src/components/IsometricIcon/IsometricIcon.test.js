import React from 'react';
import { render } from '../../../test-utils';

import IsometricIcon from './IsometricIcon';

describe('IsometricIcon', () => {
  describe('render classes as expected for IsometricIcon component', () => {
    it('it renders the IsometricIcon base class', () => {
      const { container } = render(
        <IsometricIcon id="empty-message" icon="EMPTY" />
      );

      const baseClass = container.querySelector('.pal--isometric-icon');

      expect(baseClass).toBeInTheDocument();
    });

    it('renders the error variant as expected', () => {
      const { baseElement } = render(
        <IsometricIcon id="error-message" icon="ERROR" />
      );

      expect(baseElement).toMatchSnapshot();
    });
  });
});
