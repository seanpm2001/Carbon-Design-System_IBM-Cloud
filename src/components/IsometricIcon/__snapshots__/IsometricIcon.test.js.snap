// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsometricIcon render classes as expected for IsometricIcon component renders the error variant as expected 1`] = `
<body>
  <div>
    <div
      class="cds--white cds--layer-one"
    >
      <svg
        class="pal--isometric-icon"
        viewBox="0 0 80 80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="icon--error--linear-gradient--error-message"
            x1="29.96"
            x2="53.15"
            y1="45.68"
            y2="85.84"
          >
            <stop
              offset="0"
              stop-color="#525252"
              stop-opacity=".05"
            />
            <stop
              offset="1"
              stop-opacity=".1"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="icon--error--linear-gradient-2--error-message"
            x1="38.91"
            x2="38.91"
            y1="52.59"
            y2="3.3"
          >
            <stop
              offset="0"
              stop-color="#c6c6c6"
            />
            <stop
              offset=".78"
              stop-color="#e0e0e0"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="icon--error--linear-gradient-3--error-message"
            x1="18.07"
            x2="71.64"
            y1="14.06"
            y2="44.99"
          >
            <stop
              offset="0"
              stop-color="#e0e0e0"
            />
            <stop
              offset=".13"
              stop-color="#f4f4f4"
            />
            <stop
              offset=".56"
              stop-color="#e0e0e0"
            />
            <stop
              offset=".62"
              stop-color="#d8d8d8"
            />
            <stop
              offset=".7"
              stop-color="#c6c6c6"
            />
            <stop
              offset=".89"
              stop-color="#a8a8a8"
            />
            <stop
              offset="1"
              stop-color="#8d8d8d"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="icon--error--linear-gradient-4--error-message"
            x1="27.94"
            x2="49.87"
            y1="51.19"
            y2="13.21"
          >
            <stop
              offset=".54"
              stop-color="#d0d0d0"
              stop-opacity="0"
            />
            <stop
              offset=".82"
              stop-color="#f1f1f1"
              stop-opacity=".7"
            />
            <stop
              offset=".94"
              stop-color="#fff"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="icon--error--linear-gradient-5--error-message"
            x1="28.66"
            x2="47.15"
            y1="26.33"
            y2="37"
          >
            <stop
              offset="0"
              stop-color="#fff"
            />
            <stop
              offset=".05"
              stop-color="#fdfdfd"
            />
            <stop
              offset=".3"
              stop-color="#f6f6f6"
            />
            <stop
              offset="1"
              stop-color="#f4f4f4"
            />
          </lineargradient>
        </defs>
        <path
          d="M59.91 78.34l-43-24.83 4.86-2.8 43 24.82-4.86 2.8z"
          fill="url(#icon--error--linear-gradient--error-message)"
        />
        <path
          d="M38.98 6.92C51.13 13.94 60.94 30.94 60.9 44.9s-9.92 19.6-22.07 12.58S16.87 33.46 16.91 19.5 26.83-.1 38.98 6.92z"
          fill="url(#icon--error--linear-gradient-2--error-message)"
        />
        <path
          d="M42.85 4.68C36.74 1.15 31.2.82 27.2 3.15l-3.66 2.13c3.98-2.2 9.43-1.83 15.44 1.64C51.13 13.94 60.94 30.94 60.9 44.9c-.02 6.76-2.35 11.57-6.13 13.93l-.07.05c-.32.2 3.66-2.1 3.66-2.1 3.95-2.3 6.4-7.19 6.41-14.12C64.81 28.7 55 11.69 42.85 4.68z"
          fill="url(#icon--error--linear-gradient-3--error-message)"
        />
        <path
          d="M29.11 3.9v.37a19.58 19.58 0 019.68 2.97c12.03 6.94 21.78 23.84 21.74 37.65-.02 9.4-4.56 15.23-11.83 15.23a19.59 19.59 0 01-9.68-2.96C26.99 50.22 17.24 33.32 17.28 19.5c.02-9.39 4.56-15.22 11.83-15.22V3.9m0 0c-7.2 0-12.17 5.7-12.2 15.59-.04 13.96 9.77 30.97 21.92 37.98a20.03 20.03 0 009.87 3.01c7.2 0 12.17-5.7 12.2-15.6.04-13.95-9.77-30.96-21.92-37.97A20.03 20.03 0 0029.1 3.9z"
          fill="url(#icon--error--linear-gradient-4--error-message)"
        />
        <path
          d="M38.93 49.8a6.84 6.84 0 01-2.66-2.52 6.07 6.07 0 01-.81-3v-.98a2.25 2.25 0 01.81-2.05q.81-.52 2.66.55a6.72 6.72 0 012.6 2.5 6.04 6.04 0 01.82 2.98v.99a2.25 2.25 0 01-.82 2.05c-.54.34-1.4.17-2.6-.53zm-1.16-11.64l-1.75-15.39v-9.73l5.81 3.36v9.73l-1.66 13.42z"
          fill="url(#icon--error--linear-gradient-5--error-message)"
        />
      </svg>
    </div>
  </div>
</body>
`;
